dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategySelector.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import java.util.List;
+import java.util.Optional;
+
+interface NodeSelectionStrategySelector {
```

Comment:
thoughts on calling this `NodeSelectionStrategyChooser` ? the repetition of 'selection' and 'selector' kinda messes with my head

Reply:
sure

Reply:
alternatively maybe it would actually help readability to just collapse this into the class given that we're not doing the fancy voting just yet?

Reply:
renamed